buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -409,9 +409,12 @@ class BuildPlugin implements Plugin<Project> {
         project.afterEvaluate {
             // fail on all javac warnings
             project.tasks.withType(JavaCompile) {
-                options.fork = true
-                options.forkOptions.executable = new File(project.javaHome, 'bin/javac')
-                options.forkOptions.memoryMaximumSize = "1g"
+                File gradleJavaHome = Jvm.current().javaHome
+                if (new File(project.javaHome).canonicalPath != gradleJavaHome.canonicalPath) {
+                    options.fork = true
```

Comment:
Sure. I'm fine with removing the fork. I think it is worth updating the commit message to make it clear that we no longer fork all the time, just if needed.